modules/frontend/metrics_query_range_handler.go

```diff
@@ -80,7 +84,15 @@ func newMetricsQueryRangeHTTPHandler(cfg Config, next pipeline.AsyncRoundTripper
 		logQueryRangeRequest(logger, tenant, queryRangeReq)
 
 		// build and use roundtripper
-		combiner := combiner.NewTypedQueryRange()
+		combiner, err := combiner.NewTypedQueryRange(queryRangeReq)
+		if err != nil {
+			level.Error(logger).Log("msg", "query range: query range combiner failed", "err", err)
+			return &http.Response{
+				StatusCode: http.StatusInternalServerError,
```

Comment:
Right, this is an odd situation and I was on the fence of which status code is more appropriate.  Currently it can only fail because of a query parse error which should be BadRequest. However the query has already been validated twice: in the query-frontend and also again right above this step.   Was thinking that if a new possible error condition is introduced in the future, then InternalServerError is the "safer" choice

Reply:
make sense +1